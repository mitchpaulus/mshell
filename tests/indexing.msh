[
1 [1 2 3] :2: + str
"My string" :3:
myliteral :4:
] " " join wl

"My string" :5 wl
"Another" 3: wl
"Full Range" 1:3 wl

# Negative indexing
"12345" :-1: wl  # 5
"12345" -3: wl   # 345
"12345" :-4 wl   # 1
"12345" -4:-2 wl # 23

def cjoin ([str] --) ", " join wl end

# Test deletion at index
["a" "b" "c" "d"] 2 del cjoin
1 ["a" "b" "c" "d"] del cjoin
["a" "b" "c" "d"] -1 del cjoin
-1 ["a" "b" "c" "d"] del cjoin

# Test insertion at index
["a" "b" "c" "d"] "x" 2 insert cjoin
["a" "b" "c" "d"] "x" -1 insert cjoin

# Test updating at index
["a" "b" "c" "d"] "x" 2 setAt cjoin
["a" "b" "c" "d"] "x" -1 setAt cjoin

["a" "b" "c" "d"] reverse cjoin

# Test multiple slices
["0" "1" "2" "3" "4" "5" "6" "7" "8" "9"] testList!

@testList    :2:, -1:, 3:5, :2, cjoin
"0123456789" :2:, -1:, 3:5, :2, wl
--0123456789 :2:, -1:, 3:5, :2, wl
`0123456789` :2:, -1:, 3:5, :2, str wl
[
    [printf "1\n"]
    [echo 2]
    [cat]
] | :0:, :2: ;
