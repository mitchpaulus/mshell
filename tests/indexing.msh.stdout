4 s t
My st
ther
ul
5
345
1
23
a, b, d
a, c, d
a, b, c
a, b, c
a, b, x, c, d
a, b, c, x, d
a, b, x, d
a, b, c, x
d, c, b, a
2, 9, 3, 4, 0, 1
293401
0912--
293401
1
